central/declarativeconfig/updater/auth_provider_updater.go

```diff
@@ -104,17 +104,20 @@ func (u *authProviderUpdater) DeleteResources(ctx context.Context, resourceIDsTo
 			continue
 		}
 		if err := u.authProviderRegistry.DeleteProvider(ctx, authProvider.GetId(), true, true); err != nil {
-			authProviderDeletionErr, authProviderIDs = u.processDeletionError(authProviderDeletionErr, err, authProviderIDs, authProvider)
+			authProviderDeletionErr, authProviderIDs = u.processDeletionError(ctx, authProviderDeletionErr, err, authProviderIDs, authProvider)
 		}
 	}
 	return authProviderIDs.AsSlice(), authProviderDeletionErr.ErrorOrNil()
 }
 
-func (u *authProviderUpdater) processDeletionError(authProviderDeletionErr *multierror.Error, err error, authProviderIDs set.Set[string], authProvider *storage.AuthProvider) (*multierror.Error, set.Set[string]) {
+func (u *authProviderUpdater) processDeletionError(ctx context.Context, authProviderDeletionErr *multierror.Error,
+	err error, authProviderIDs set.Set[string], authProvider *storage.AuthProvider) (*multierror.Error, set.Set[string]) {
 	authProviderDeletionErr = multierror.Append(authProviderDeletionErr, err)
 	authProviderIDs.Add(authProvider.GetId())
 
-	u.reporter.UpdateIntegrationHealthAsync(declarativeCfgUtils.IntegrationHealthForProtoMessage(authProvider, "", err,
-		u.idExtractor, u.nameExtractor))
+	if err := u.healthDS.UpsertDeclarativeConfig(ctx, declarativeCfgUtils.HealthStatusForProtoMessage(authProvider, "", err,
```

Comment:
Also losing `handler` value here, same as in notifier. Can be solved in a follow up though